modules/ROOT/pages/UploadToSMR.adoc

```diff
@@ -1,55 +1,107 @@
+**You can't change your mod reference after you have added your mod to SMR!**
 
-So make sure it is correct! At least five times! And read up on the xref:Development/BeginnersGuide/index.adoc#_mod_reference[Mod Reference section] again to make sure you understand what other parts of your mod need to use this same value.
+So make sure it is correct! At least five times!
+And read up on the xref:Development/BeginnersGuide/index.adoc#_mod_reference[Mod Reference section]
+again to make sure you understand what other parts of your mod need to use this same value.
 ====
 
-Next, you can upload an image to use as the display image for your mod,
-and a link to your source code if it is hosted on a site like GitHub or similar.
-We strongly encourage you to release your source code, as it allows others
-to learn from your work and helps other developers debug mod compatibility.
+=== Icon and Source Link
+
+Next, you can upload an image to use as the display image for your mod.
+Alternatively, you can leave this blank, and it will default to the Satisfactory Modding gear logo.
+
+You can also provide a link to your source code, if it is hosted on a site like GitHub or similar.
+We strongly encourage you to release your source code,
+as it allows others to learn from your work
+and helps other developers debug mod compatibility.
+
+You can edit both of these fields later.
+
+=== Hidden Mods
+
+You also have the option to mark your mod as Hidden upon creation.
+You can go back and edit this later.
+
+If your mod is Hidden, it will not appear in the main mods list,
+nor when browsing in SMM,
+nor when searching the site.
+However, other users can still find it by looking at your profile page,
+and you can send people the link directly.
+
+In addition, updates to Hidden mods will not cause messages to be sent in the `#mod-updates` channel of the modding Discord.
+
+=== Submit your Mod Page
+
+Once you click Submit, a page for your mod will be created.
+
+Note that the page for your mod contains your mod's mod reference as part of the URL.
+For example, the page for a mod with reference `AreaActions` would be `https://ficsit.app/mod/AreaActions`.
 
-Once you click Submit, your mod page will be created.
 You still need to upload a version for your mod to be usable by players.
-Until then, your mod will be marked as Outdated since there are no compatible versions.
+Until then, your mod will be marked as 'Outdated' since there are no compatible versions.
 This version submission process will be covered in the following sections.
 
-If you'd like, you can click 'Edit Mod' and hide your mod from discovery.
-This is helpful for making a library mod, for example,
-or if you need to split your mod into multiple sub-mods.
+== Your Mod's UPlugin File
 
-== UPlugin File
+Before you can upload a version,
+there are a few fields you need to edit in your mod's uplugin file.
 
 Every Unreal Engine plugin is require to have a
```

Comment:
Do we explain the concept of .uplugin files elsewhere? If we do, we should probably just link that instead I think. This doesn't really fit the goal of the page

Reply:
It looks like the closest we get is telling people to write stuff in some fields in the UE plugin editing GUI in the Plugin Creation docs page.

I think that if these fields were excluded, your mod would still probably work (outside of the mod loading library calls to check its semversion and such)

Reply:
Mh, I see. Let's maybe make a not that we should have a dedicated uplugin page?

Reply:
Done in another reply chain